clang/lib/CodeGen/ItaniumCXXABI.cpp

```diff
@@ -4207,16 +4214,31 @@ llvm::Constant *ItaniumRTTIBuilder::BuildTypeInfo(
     break;
   }
 
-  llvm::Constant *Init = llvm::ConstantStruct::getAnon(Fields);
-
   SmallString<256> Name;
   llvm::raw_svector_ostream Out(Name);
   CGM.getCXXABI().getMangleContext().mangleCXXRTTI(Ty, Out);
   llvm::Module &M = CGM.getModule();
   llvm::GlobalVariable *OldGV = M.getNamedGlobal(Name);
-  llvm::GlobalVariable *GV =
-      new llvm::GlobalVariable(M, Init->getType(),
-                               /*isConstant=*/true, Linkage, Init, Name);
+  llvm::GlobalVariable *GV = new llvm::GlobalVariable(
+      M, llvm::ConstantStruct::getTypeForElements(Fields),
```

Comment:
Since you're going to replace the initializer anyway, you don't need to call ConstantStruct::getTypeForElements here; you can just use an arbitrary type, like i8.  That should allow you to pass the global into BuildVTablePointer(), instead of replacing it later.

Reply:
@efriedma-quic Thanks for suggestion! Followed this way in b08172fa576e5bb7849aa13d503180cfef5bcd0b. This actually changes order of globals definitions in output IR and requires changing related tests correspondingly (see c9d26dd6dd7648c25e44e576891e838a418eacae), but, as far as I understand, this should not cause any problems.